qiskit/circuit/library/overlap.py

```diff
@@ -0,0 +1,61 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Unitary overlap circuit."""
+from qiskit.circuit import QuantumCircuit
+from qiskit.circuit.parametervector import ParameterVector
+from qiskit.circuit.exceptions import CircuitError
+
+
+class UnitaryOverlap(QuantumCircuit):
+    r"""Circuit that returns the overlap between two unitaries $V^{\dag}U|0\rangle^{\otimes N}$.
+
+    Input quantum circuits must represent unitary operations that are invertible.  Input
+    circuits `U` and `V` will have parameters, if any, renamed `ParameterVector` `a` and `b`,
+    respectively in the output circuit.
+
+    Circuit is usually employed in computing the fidelity
+    $\left|\langle 0\dots 0| V^{\dag}U|0\dots 0\rangle\right|^{2}$ by computing the probability
+    of being in the all-zeros bit-string, or equivalently, the expectation value of projector
+    $|0\dots 0\rangle\langle 0\dots 0|$.
+    """
```

Comment:
Afaik we have to use the `:math:` environment for TeX to be rendered correctly 🙂 
```suggestion
    r"""Circuit that returns the overlap between two unitaries :math:`V^\dagger U|0\rangle`.

    Input quantum circuits must represent unitary operations that are invertible. Parameters
    in the circuits ``U`` and ``V`` will be replaced by a :class:`.ParameterVector` with name
    ``"a"`` and ``"b"`` in the output circuit.

    A unitary overlap circuit is usually employed in computing the fidelity
    :math:`|\langle 0| V^\dagger U |0\rangle|^2` by computing the probability
    of being in the all-zeros bit-string, or equivalently, the expectation value of projector
    :math:`|0\rangle\langle 0|` acting on all qubits.
    """
```